models/classes/LtiProvider/LtiProvider.php

```diff
@@ -45,99 +47,145 @@ class LtiProvider implements JsonSerializable
     /** @var array */
     private $roles;
 
-    /**
-     * @param string $id
-     * @param string $label
-     * @param string $key
-     * @param string $secret
-     * @param string $callbackUrl
-     * @param string[] $roles
-     */
-    public function __construct($id, $label, $key, $secret, $callbackUrl, array $roles = [])
-    {
+    /** @var string */
+    private $ltiVersion;
+
+    /** @var string */
+    private $toolClientId;
+
+    /** @var string */
+    private $toolAudience;
+
+    /** @var string[] */
+    private $toolDeploymentIds = [];
+
+    /** @var string */
+    private $toolOidcLoginInitiationUrl;
+
+    /** @var string */
+    private $toolLaunchUrl;
+
+    /** @var string */
+    private $toolPublicKey;
+
+    /** @var string */
+    private $toolIdentifier;
+
+    /** @var string */
+    private $toolName;
+
+    /** @var string */
+    private $toolJwksUrl;
+
+    public function __construct(
+        string $id,
+        string $label,
+        string $key,
+        string $secret,
+        string $callbackUrl,
+        array $roles = [],
+        string $ltiVersion = null,
```

Comment:
About the long parameters list in the construct, the only way I see to fix it is creating different objects, but it would imply to change all the places using this class. I think this is not feasible at this point. Wdyt?

Reply:
lets just create a improvement list , but i am with this for the moment